tests/tests.py

```diff
@@ -55,6 +55,14 @@ def test_multi_super(self):
             result, "block2 from test6 - block2 from test3 - block2 from test1"
         )
 
+    def test_render_block_in_multi_inherited_template(self):
+        result = render_block_to_string("test8_django.html", "block1")
+        self.assertEqual(result, "block3 from test8")
+
+    def test_multi_level_inherited_template_block(self):
+        result = render_block_to_string("test8_django.html", "block1")
+        self.assertEqual(result, "block3 from test8")
```

Comment:
These two tests look identical to me? Is there a difference?

Reply:
Oh sure. I removed that test. I was trying to test two different bugs, but it is covered by that one test. 

1st issue:
Error:
`render_block.exceptions.BlockNotFound: block with name 'block1' does not exist`
Solved by iterating through all parent templates until block is found and rendered.

2nd issue:
There was workaround to fix 1st issue by adding `{% block block1 %}{{ block.super }}{% endblock %}` to  test8_django.html, but when we rendered block1 with test8_django.html template, it was rendering block3 from test7 and not block3 from test8
It is resolving blocks in similar way as django's `ExtendsNode` render implementation: https://github.com/django/django/blob/main/django/template/loader_tags.py#L141

Reply:
Thanks for clarifying! Will try to review soon.